micro_ros_diagnostic_updater/include/micro_ros_diagnostic_updater/micro_ros_diagnostic_updater.h

```diff
@@ -22,42 +22,61 @@
 
 #define MICRO_ROS_UPDATER_MAX_NUMBER_OF_TASKS 5
 
-typedef struct diagnostic_key_value_t
+typedef struct diagnostic_value_t
 {
-  const char * key;
-  const char * value;
-} diagnostic_key_value_t;
+  int8_t value_type;
+  bool bool_value;
+  int16_t int_value;
+  double double_value;
```

Comment:
As this is intended to run on an embedded system, wouldn't it be better to avoid bringing in doubles? rather prefer using float(4 bytes).

On a side note, on ARM chips, if you can avoid using floats/doubles at all, you can save up to 10kB on libraries required for floating-point operations. Maybe considering giving the user the opportunity to decide not to bring them at all would be nice.